cli/mender-artifact/read.go

```diff
@@ -75,20 +75,22 @@ func readArtifact(c *cli.Context) error {
 	}
 
 	ar := areader.NewReader(f)
-	r, err := read(ar, ver, readScripts)
+	ar.ScriptsReadCallback = readScripts
```

Comment:
Suggestion: Maybe `NewReader` should accept the callback functions?

Reply:
How about nuking them altogether? X-D

To save effort, I think I'll leave this as it is, we both know this area needs serious rewriting.

Reply:
> How about nuking them altogether? X-D

That would be the preferred option yes! 

> 
> To save effort, I think I'll leave this as it is, we both know this area needs serious rewriting.

Yeh, It's nothing major.